pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -3102,3 +3110,53 @@ func handleDeviceOperation(ctxPtr *zedagentContext, op types.DeviceOperation) {
 	shutdownAppsGlobal(ctxPtr)
 	// nothing else to be done
 }
+
+func parseEdgeNodeClusterConfig(getconfigCtx *getconfigContext,
+	config *zconfig.EdgeDevConfig) {
+
+	ctx := getconfigCtx.zedagentCtx
+	zcfgCluster := config.GetCluster()
+	if zcfgCluster == nil {
+		log.Functionf("parseEdgeNodeClusterConfig: No EdgeNodeClusterConfig, Unpublishing")
+		pub := ctx.pubEdgeNodeClusterConfig
+		items := pub.GetAll()
+		if len(items) > 0 {
+			log.Functionf("parseEdgeNodeClusterConfig: Unpublishing EdgeNodeClusterConfig")
+			ctx.pubEdgeNodeClusterConfig.Unpublish("global")
+		}
+		return
+	}
+	ipAddr, ipNet, err := net.ParseCIDR(zcfgCluster.GetClusterIpPrefix())
+	if err != nil {
+		log.Errorf("handleEdgeNodeConfigItem: ParseCIDR failed %s", err)
```

Comment:
handleEdgeNodeConfigItem -> parseEdgeNodeClusterConfig

Reply:
Sure.


> handleEdgeNodeConfigItem will be submitted in another PR?

No. the function was used during the testing of cluster without controller side support, thus it was a temp function to handle cluster config and cluster-wide pubsub. We don't need this anymore.